File: app/src/main/java/com/iceteaviet/fastfoodfinder/utils/NetworkUtils.java
package com.iceteaviet.fastfoodfinder.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import java.io.IOException;

/**
 * Created by Genius Doan on 22/07/2017.
 */

public class NetworkUtils {
    public static boolean isNetworkReachable(Context context) {
        ConnectivityManager conMgr = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = conMgr.getActiveNetworkInfo();

        if (networkInfo != null && networkInfo.isConnected()) {
            return true;
        } else {
            return false;
        }
    }

    public static boolean isInternetConnected() {
        String command = "ping -c 1 google.com";
        try {
            return (Runtime.getRuntime().exec(command).waitFor() == 0);
        } catch (InterruptedException | IOException e) {
            e.printStackTrace();
            return false;
        }
    }
}

File: app/src/main/java/com/iceteaviet/fastfoodfinder/utils/NetworkUtils.kt
@file:JvmName("NetworkUtils")

package com.iceteaviet.fastfoodfinder.utils

import android.content.Context
import com.iceteaviet.fastfoodfinder.androidext.getConnectivityManager
import com.iceteaviet.fastfoodfinder.utils.shell.AppShell
import java.io.IOException

/**
 * Created by Genius Doan on 22/07/2017.
 */

/**
 * Check internet connection access
 */
fun isInternetConnected(): Boolean {
    val command = "ping -c 1 google.com"
    try {
        return AppShell.exec(command).waitFor() == 0
    } catch (se: SecurityException) {
        return false
    } catch (ioEx: IOException) {
        return false
    } catch (interruptException: InterruptedException) {
        return false
    }
}

/**
 * Check connected to any network (but don't need to have Internet access)
 */
fun isNetworkReachable(context: Context): Boolean {
    val conMgr = context.getConnectivityManager()
    val networkInfo = conMgr?.activeNetworkInfo

    return networkInfo != null && networkInfo.isConnected
}
